sycl/doc/extensions/proposed/sycl_ext_oneapi_graph.asciidoc

```diff
@@ -209,20 +519,292 @@ node add_malloc_device(void *&data, size_t numBytes, const std::vector<node>& de
 ----
 |Adding a node that encapsulates a `malloc` operation.
 
+Parameters:
+
+* `data` - Return parameter set to the address of memory allocated.
+
+* `numBytes` - Size in bytes to allocate.
+
+* `dep` - Nodes the created node will be dependent on.
+
+Returns: The memory allocation node which has been added to the graph
+
+Exceptions:
+
+* Throws synchronously with error code `invalid` if a queue is recording
+  commands to the graph.
+
 |
 [source,c++]
 ----
 node add_free(void *data, const std::vector<node>& dep = {});
 ----
 |Adding a node that encapsulates a `free` operation.
 
+Parameters:
+
+* `data` - Address of memory to free.
+
+* `dep` - Nodes the created node will be dependent on.
+
+Returns: The memory freeing node which has been added to the graph.
+
+Exceptions:
+
+* Throws synchronously with error code `invalid` if a queue is recording
+  commands to the graph.
+
+|===
+
+Table 8. Member functions of the `command_graph` class (executable graph update).
+[cols="2a,a"]
 |===
+|Member function|Description
+
+|
+[source, c++]
+----
+void command_graph<graph_state::executable> update(const command_graph<graph_state::modifiable> &graph);
+----
+
+|Updates the executable graph node inputs & outputs from a topologically
+identical modifiable graph. The effects of the update will be visible
+on the next submission of the executable graph without the need for additional
+user synchronization.
+
+Parameters:
+
+* `graph` - Modifiable graph object to update graph node inputs & outputs with.
+  This graph must have the same topology as the original graph used on
+  executable graph creation.
+
+Preconditions:
+
+* This member function is only available when the `command_graph` state is
+  `graph_state::executable`.
+
+Exceptions:
+
+* Throws synchronously with error code `invalid` if the topology of `graph` is
```

Comment:
Do we need more specific errors? A whole-graph update seems error-prone and hard to debug to me.

Reply:
When we looked through the SYCL spec the wording around what's thrown on errors tends to be relatively sparse and limited to error codes. My understanding is that the runtime should put more helpful error information in the `what()` message of the exception, but I don't think we need to specify exactly what form that returned string will take.

Reply:
Going to resolve this thread since the current wording is what we have in the Codeplay spec, and I don't want to expand the scope of this PR too much from just merging the two specs. However, if you want to discuss it further please make an Issue for it, because it is a good question.